src/fhirtypes/ElementDefinition.ts

```diff
@@ -820,15 +956,35 @@ export class ElementDefinition {
     if (isEmpty(validTypes)) {
       throw new CodedTypeNotFoundError(this.type ? this.type.map(t => t.code) : []);
     }
-
+    const strengths = ['example', 'preferred', 'extensible', 'required'];
     // Check if this is a valid strength (if the binding.strength already exists)
     if (this.binding && this.binding.strength) {
-      const strengths = ['example', 'preferred', 'extensible', 'required'];
       if (strengths.indexOf(strength) < strengths.indexOf(this.binding.strength)) {
         throw new BindingStrengthError(this.binding.strength, strength);
       }
     }
 
+    const connectedElements = this.findConnectedElements();
+    connectedElements.forEach(ce => {
+      if (vsURI == ce.binding?.valueSet) {
+        try {
+          ce.bindToVS(vsURI, strength);
```

Comment:
Is it overly complicated to use `bindToVs` recursively here? Since you are not actually using the error checking that this function provides, does it make more sense to just set the `vsURI` and `strength` directly on `ce.binding`? Or is there something wrong with that approach?

Reply:
The advantage of calling `bindToVS` recursively is that it already performs the binding strength checks that I want to perform. If I didn't do that, I'd need to duplicate those checks inside the `try` block.